foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/actions/FilterAction.java

```diff
@@ -107,22 +112,23 @@ public void validateImpl(Query parameter) {
             validationErrors.add("limit must be positive integer");
         }
 
+        if (parameter.getLimit() > elasticsearchTuningConfig.getDocumentsLimitAllowed()){
+            validationErrors.add(String.format("Limit more than %s is not supported",
+                                               elasticsearchTuningConfig.getDocumentsLimitAllowed()));
+        }
+
         if(!CollectionUtils.isNullOrEmpty(validationErrors)) {
             throw FoxtrotExceptions.createMalformedQueryException(parameter, validationErrors);
         }
     }
 
     @Override
     public ActionResponse execute(Query parameter) {
-        SearchRequest search = getRequestBuilder(parameter);
-        try {
-            logger.info("Search: {}", search);
-            SearchResponse response = getConnection()
-                .getClient()
-                .search(search);
-            return getResponse(response, parameter);
-        } catch (IOException e) {
-            throw FoxtrotExceptions.createQueryExecutionException(parameter, e);
+        if (parameter.isScrollRequest()) {
+            return executeScrollRequest(parameter);
+        }
+        else {
```

Comment:
Remove else and let this be the default path. Looks cleaner. Compiler will add the else for you.